mantidproject/mantid
# GSASIIRefineFitPeaks enable refining sigma and gamma

Refining sigma and gamma profile coefficients was enabled in GSASIIRefineFitPeaks.

**To test:**

Make sure the system test for GSASIIRefineFitPeaks passes. This is NOT run on the build servers. 

You'll need GSAS-II installed. Install it using one of the scripts in `scripts/GSAS-II` or follow the instructions [here](https://subversion.xray.aps.anl.gov/trac/pyGSAS). Test data is in [GSASIIRefineFitPeaks.zip](https://github.com/mantidproject/mantid/files/1739753/GSASIIRefineFitPeaks.zip)


```
import os

# Replace this with your gsas-ii installation directory
path_to_gsas = "/directory/containing/GSASIIscriptable.py"

# Replace this with a location to save the GSAS-II project file
gsas_proj_file = "/somewhere/local/Test.gpx"

data_dir = "/wherever/you/extracted/test/data"
input_file = lambda file: os.path.join(data_dir, file)

phase_files = ",".join([input_file("Fe-{}.cif".format(phase)) for phase in ["alpha", "gamma", "epsilon"]])
iparams_file = input_file("template_ENGINX_241391_236516_North_bank.prm")
input_ws = Load(Filename=input_file("ENGINX_280625_focused_bank_1.nxs"))

fitted_peaks, lattice_params, rwp, sigma, gamma = \
    GSASIIRefineFitPeaks(InputWorkspace=input_ws,
                         PhaseInfoFiles=phase_files,
                         InstrumentFile=iparams_file,
                         RefinementMethod="Rietveld refinement",
                         SaveGSASIIProjectFile=gsas_proj_file,
                         PathToGSASII=path_to_gsas,
                         RefineSigma=True,
                         RefineGamma=True,
                         MuteGSASII=True) # Take this out if you're interested in GSAS-II output

print("rwp = {}".format(rwp))
print("sigma = {}".format(sigma))
print("gamma = {}".format(gamma))
```

You'll get a warning about one of the phase files (unrelated to this PR), and hopefully the following output:
```
rwp = 48.2506194254
sigma = 1245.75079453
gamma = 81.2269608793
```
![capture](https://user-images.githubusercontent.com/29916298/36419499-ffe09484-1629-11e8-81c3-4165a35bae53.PNG)

Try turning sigma or gamma off, and make sure the output workspace is different.

Fixes #21575 

**Release Notes** 
Not in the release notes, as it's part of the general work to bring GSASIIRefineFitPeaks up to scratch

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
